### PR TITLE
update react resources page

### PR DESCRIPTION
**Pull request by k-m-crawford**

### CODE REVIEW

File: Topics/Tech_Stacks/React.md [hunk 1]
```diff
@@ -1,37 +1,59 @@
...
+A free course provided by Scrimba that teaches React fundamentals through the creation of various applications including an AirBnB clone, a meme generator, and a notes app.
 
-React tutorial by MDN Web Docs. Useful readings for if you have time, but not necessary by any means.\
-https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started
+[Official React Tutorial](https://reactjs.org/tutorial/tutorial.html)
+Basic React tutorial provided by the official React website. This tutorial will guide you through a very basic tutorial that you can do straight from your browser. 
 
+[React JS Crash Course](https://www.youtube.com/watch?v=w7ejDZ8SWv8)
+[One Hour React Tutorial](https://www.youtube.com/watch?v=b9eMGE7QtTk)
+One hour in-depth tutorials about React. These Youtube crash course provides a great overview of React and various important components that you will be using often. Very useful if you are learning React under a strict time limit for whatever reason.
```
Comment (ericjest) [inline Topics/Tech_Stacks/React.md:24]: "One hour" and "in-depth" are conflicting descriptors, it might make more sense if we don't call it in-depth.

Comment (ericjest) [inline Topics/Tech_Stacks/React.md:24]: "Youtube" should be spelled as "YouTube".

File: Topics/Tech_Stacks/React.md [hunk 2]
```diff
@@ -1,37 +1,59 @@
...
 
-Javascript frameworks tutorial provided by Microsoft. Useful readings, but not necessary to learn React. \
-https://learn.microsoft.com/en-us/windows/dev-environment/javascript/
+[Front-End Mentor](https://www.frontendmentor.io/)
+These frontend challenges for HTML, CSS and JavaScript will help you review core skills while also making interesting projects.
 
+[Mozilla React Tutorial](https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Client-side_JavaScript_frameworks/React_getting_started)
```
Comment (ericjest) [inline Topics/Tech_Stacks/React.md:29]: MDN is excellent. I would strongly advocate adding the official React docs to this section, imo it is the best place for a beginner to learn React. But up to you

File: Topics/Tech_Stacks/React.md [hunk 3]
```diff
@@ -1,37 +1,59 @@
...
+React tutorial by MDN Web Docs. Useful readings for if you have time, but not necessary by any means.
 
-1 hour in depth tutorials about React. These Youtube crash course provides a great in depth look at React and the various important components that you will be using most often. Very useful if you are learning React under a strict time limit for whatever reason. \
-https://www.youtube.com/watch?v=w7ejDZ8SWv8 \
-https://www.youtube.com/watch?v=b9eMGE7QtTk 
+[Microsoft JavaScript Tutorial](https://learn.microsoft.com/en-us/windows/dev-environment/javascript/)
+Javascript frameworks tutorial provided by Microsoft. Useful readings, but not necessary to learn React.
```
Comment (ericjest) [inline Topics/Tech_Stacks/React.md:33]: Javascript should be spelled "JavaScript"

File: Topics/Tech_Stacks/React.md [hunk 4]
```diff
@@ -1,37 +1,59 @@
...
+[Material UI](https://mui.com/material-ui/)
+A fully-featured UI library that offers a huge range of common components that are easily styled and based on Google's Material Design.
+
+[Bootstrap](https://getbootstrap.com/)
+Another responsive UI library that allows you to focus on development by handling the design logistics for you.  
+
+### Miscellaneous
+
+[GitHub Pages](https://pages.github.com/)
+Host your React websites straight from your Github repo. Website hosting is fairly expensive, so this is here as a free alternative for website deployment for simple applications.
```
Comment (ericjest) [inline Topics/Tech_Stacks/React.md:55]: I'm nitpicking but I don't think website hosting is too expensive, so maybe we can reword this to just say "If you don't want to spend money on website hosting" or "If you don't need server functionality and want to host for free" or something like that.

**Review by ericjest — CHANGES_REQUESTED** [review]

File: Topics/Tech_Stacks/React.md (hunk 4 above)
Comment (k-m-crawford) [reply Topics/Tech_Stacks/React.md:55]: Fair point

File: Topics/Tech_Stacks/React.md (hunk 2 above)
Comment (k-m-crawford) [reply Topics/Tech_Stacks/React.md:29]: true, added those as well.

**Review by ericjest — APPROVED** [review]

**Comment by ericjest** [comment]
Looks good to me

File: Topics/Tech_Stacks/React.md [hunk 5]
```diff
@@ -1,37 +1,62 @@
 ## Tech Stacks
 
 ### React Overview
-React is a Javascript library used to build user interfaces. As such, you're going to have to know HTML, CSS and Javascript in order to learn React.
-It's ease of use and popularity comes from how little you actually have to code. Easily applicable for any frontend interface, if you're going to use any library/framework for frontend development, React is the safest option.
+React is an open-source JavaScript library for building component-based user interfaces. This component-based architecture allows engineers to quickly prototype and develop complex applications easily by modularizing common functionalities into discrete, reusable UI components, and React boasts a large user base full of freely available components libraries. Beyond this, React there's much more to learn about React, such as its virtual DOM, JSX, and a full mobile framework in React Native, and much more -- so much more that it can be [overwhelming](https://auth0.com/blog/how-to-manage-javascript-fatigue/).
```
Comment (ICPRplshelp) [inline Topics/Tech_Stacks/React.md:4]: There seems to be some typos. Replace this entire section with this
```suggestion
React is an open-source JavaScript library for building component-based user interfaces. This component-based architecture allows engineers to quickly prototype and develop complex applications easily by modularizing common functionalities into discrete, reusable UI components, whilst boasting a large user base full of freely available component libraries. Beyond this, there's much more to learn about React, such as its virtual DOM, JSX, a full mobile framework in React Native, and much more -- so much more that it can be [overwhelming](https://auth0.com/blog/how-to-manage-javascript-fatigue/). 
```

File: Topics/Tech_Stacks/React.md (hunk 4 above)
Comment (ICPRplshelp) [reply Topics/Tech_Stacks/React.md:58]: It might be a good idea to plug the page I contributed (GitHub pages). I am a bit unsure on how to do that as it is not committed yet.

File: Topics/Tech_Stacks/React.md [hunk 6]
```diff
@@ -1,37 +1,62 @@
...
+React tutorial by MDN Web Docs. Useful readings for if you have time.
 
+[Microsoft JavaScript Tutorial](https://learn.microsoft.com/en-us/windows/dev-environment/javascript/)
+JavaScript frameworks tutorial provided by Microsoft. Useful readings, but not necessary to learn React. 
 
-Online development environment to try React out in (Among other frontend tools). Great for testing the waters with React to see if it will be something you want to learn or not. \
-https://codepen.io/
+### Useful React Tools and Libraries
 
+[Create React App](https://create-react-app.dev/)
```
Comment (ICPRplshelp) [inline Topics/Tech_Stacks/React.md:40]: Create React App is criticized for being slow, in terms of the actual react website you make being slow. A substitute would be Vite, which works very similar to create react app. Consider putting this in.

I don't know if Next.js should be mentioned or is already mentioned here.

**Review by ICPRplshelp — CHANGES_REQUESTED** [review]
I've reviewed this based on the files you have right now. Significant changes aren't needed -- this is mostly me just suggesting some additional resources (that are free).

File: Topics/Tech_Stacks/React.md [hunk 7]
```diff
@@ -1,37 +1,62 @@
...
+[Bootstrap](https://getbootstrap.com/)
+Another responsive UI library that allows you to focus on development by handling the design logistics for you.  
+
+### Miscellaneous
+
+[GitHub Pages](https://pages.github.com/)
+Host your React websites straight from your Github repo. Website hosting is fairly expensive, so this is here as a free alternative for website deployment for simple applications.
+
+[Codepen](https://codepen.io/)
+Online development environment to try React out in (among other frontend tools). Great for testing the waters with React and other libraries or languages.
```
Comment (ICPRplshelp) [inline Topics/Tech_Stacks/React.md:61]: Consider adding Stackblitz.

**Review by ICPRplshelp — COMMENTED** [review]
Appended to the previous review

File: Topics/Tech_Stacks/React.md (hunk 4 above)
Comment (k-m-crawford) [reply Topics/Tech_Stacks/React.md:58]: I'll add a link to the index page, once you commit I can add the link if there's time.
